src/big/big_int.cr

```diff
@@ -403,6 +402,269 @@ struct BigInt < Int
     LibGMP.sizeinbase(self, 2).to_i
   end
 
+  # Mutating +
+  def add!(other : BigInt) : BigInt
+    if @mpz == other.@mpz
+      mul!(2)
+    else
+      LibGMP.add(mpz, self, other)
+    end
+    self
+  end
+
+  # Mutating +
+  def add!(other : Int) : BigInt
+    if other < 0
+      sub!(other.abs)
+    elsif other <= LibGMP::ULong::MAX
+      LibGMP.add_ui(mpz, self, other)
+      self
+    else
+      add!(other.to_big_i)
+    end
+  end
+
+  # Mutating -
+  def sub!(other : BigInt) : BigInt
+    if @mpz == other.@mpz
+      LibGMP.set_ui(mpz, 0)
+    else
+      LibGMP.sub(mpz, self, other)
+    end
+    self
+  end
+
+  # Mutating -
+  def sub!(other : Int) : BigInt
+    if other < 0
+      add!(other.abs)
+    elsif other <= LibGMP::ULong::MAX
+      LibGMP.sub_ui(mpz, self, other)
+      self
+    else
+      sub!(other.to_big_i)
+    end
+  end
+
+  def abs! : BigInt
+    LibGMP.abs(mpz, self)
+    self
+  end
+
+  # Mutating unary -
+  def neg! : BigInt
+    LibGMP.neg(mpz, self)
+    self
+  end
+
+  def factorial! : BigInt
+    if self < 0
+      raise ArgumentError.new("Factorial not defined for negative values")
+    elsif self > LibGMP::ULong::MAX
+      raise ArgumentError.new("Factorial not supported for numbers bigger than 2^64")
+    end
+    LibGMP.fac_ui(mpz, self)
+    self
+  end
+
+  # Mutating *
```

Comment:
```suggestion
  # Mutating `#*`.
```

That should make it link to like https://crystal-lang.org/api/master/BigInt.html#*(other:Int):BigInt-instance-method.

Reply:
How do I make it tell apart `-` vs. unary `-`?

Reply:
Like `#-` (unary) vs. `#-(other)` (binary)